milli/src/search/facet/grammar.pest

```diff
@@ -8,6 +8,9 @@ char =  _{ !(PEEK | "\\") ~ ANY
     | "\\" ~ (PEEK | "\\" | "/" | "b" | "f" | "n" | "r" | "t")
     | "\\" ~ ("u" ~ ASCII_HEX_DIGIT{4})}
 
+reserved = { "_geo" | "_geoDistance" | "_geoPoint" | ("_geoPoint" ~ parameters) }
+// we deliberately choose to allow empty parameters to generate more specific error message later
```

Comment:
Maybe using nom as the parser would help increase this, instead of unconstraining the pest parser to catch errors on the Rust side.

Reply:
Yes, it'll be more verbose but it should be better in the end! :grin: